src/modules/payload_deliverer/payload_deliverer.cpp

```diff
@@ -0,0 +1,255 @@
+ *
+ *   Copyright (c) 2022 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+#include "payload_deliverer.h"
+
+PayloadDeliverer::PayloadDeliverer()
+	: ModuleParams(nullptr),
+	  ScheduledWorkItem(MODULE_NAME, px4::wq_configurations::lp_default)
+{
+}
+
+bool PayloadDeliverer::init()
+{
+	// Register callback that triggers the Run() function
+	if (!_vehicle_command_sub.registerCallback()) {
+		PX4_ERR("callback registration failed");
+		return false;
+	}
+
+	initialize_gripper();
+	return true;
+}
+
+bool PayloadDeliverer::initialize_gripper()
+{
+	// If gripper instance is invalid, try initializing it
+	if (!_gripper.is_valid() && _param_gripper_enable.get()) {
+		GripperConfig config{};
+		config.type = (GripperConfig::GripperType)_param_gripper_type.get();
+		config.sensor = GripperConfig::GripperSensorType::NONE; // Feedback sensor isn't supported for now
+		config.timeout_us = hrt_abstime(_param_gripper_timeout_s.get() * 1000000ULL);
+		_gripper.init(config);
+	}
+
+	// NOTE: Support for changing gripper sensor type / gripper type configuration when
+	// the parameter change is detected isn't added as we don't have actual use case for that
+	// yet!
+
+	if (!_gripper.is_valid()) {
+		PX4_DEBUG("Gripper object initialization invalid!");
+		return false;
+
+	} else {
+		_gripper.grab(); // Initialize to grab position
```

Comment:
This is already done in the gripper init. I would remove it there.

Reply:
Makes sense!